### PR TITLE
Flag the group's own snapshot entry when a whole-group delete removes it

### PR DESCRIPTION
**Pull request by nkdengineer**
### Explanation of Change

Deleting every expense in a group made the group row flash: it disappeared, came back, then disappeared again.

A grouped search reads from **two** Onyx keys. The main snapshot holds one `group_<id>` entry per group and nothing else — [`getMemberSections` builds one row per group entry](https://github.com/Expensify/App/blob/19582b148a4/src/libs/SearchUIUtils.ts#L3582-L3590) with `transactions: []`. Each group's children live under a **separate hash**, [pulled in by `useSearchSnapshot`](https://github.com/Expensify/App/blob/19582b148a4/src/components/Search/hooks/useSearchSnapshot.ts#L259-L290) via `useMultipleSnapshots`.

The group row had no `pendingAction` of its own — it derived one from its children, and that derivation required `transactions.length > 0`. Since the two keys are cleared at different times, that produced three states:

1. All children pending delete → group derives `DELETE` → [`OfflineWithFeedback` hides the row's content](https://github.com/Expensify/App/blob/19582b148a4/src/components/OfflineWithFeedback.tsx#L98). **Row gone.**
2. Delete succeeds, children cleared from the sub-snapshot → `transactions.length === 0`, nothing left to derive from → `pendingAction` is `undefined`. **Row back — the flash.**
3. The next `Search` response drops the group entry. **Row gone for good.**

The fix moves the delete state onto the group's own snapshot entry, which is the only thing that outlives its children:

- `pendingAction` lives on a shared `SearchGroupBase` in [`SearchResults.ts`](https://github.com/Expensify/App/blob/19582b148a4/src/types/onyx/SearchResults.ts#L133-L146), which also folds up the `count`/`total`/`currency` that all ten group models repeated.
- `getMemberSections` already spreads the group entry into the row, so `item.pendingAction` lights up and the existing `isRowDeleted` checks just work. The header greys out from that flag only — there is no children-derived pending-delete fallback.

Which groups get that flag is **coverage**, not "clicked the group checkbox":

- `isSelectedViaGroup` still means the user selected through the group header. Export uses it so a group-row click can export as a group even when `limit:` left children unloaded.
- Delete uses `getSelectedGroupKeys`, which includes a group only when the selected children cover the group's snapshot `count` (or, when count is missing, when `isEntireGroupSelected` is set). Selecting every loaded child one by one is enough when that loaded set *is* the whole group. Clicking the group checkbox with `limit:` Y < X is not.
- `stampGroupCoverageFlags` writes `isEntireGroupSelected` without touching `isSelectedViaGroup`.

A `limit:` smaller than the group count only deletes the loaded rows, so the group stays in the list. In that case:

- The group is **not** flagged pending-delete, so remaining expenses stay visible.
- If the group was expanded, it [auto-collapses](https://github.com/Expensify/App/blob/19582b148a4/src/components/Search/SearchList/ListItem/shouldCollapseExpandedGroupAfterPendingDelete.ts) once every loaded child is pending-delete, instead of sitting open on an empty/deleted child list. Expanding again fetches the remaining expenses.
- The header greys out from the group's own `pendingAction` only, so a truncated delete does not grey out a group that still has expenses.

The flag rides in the delete request's own `optimisticData`/`failureData`, so a failed delete puts the row back. Delete runs once per transaction, so all of a group's requests carry the same idempotent merge and any one failing restores the row.

No `successData` is needed: the `Search` response replaces the whole snapshot, so a surviving group loses the flag and a deleted group loses its entry.

The view drops flagged rows from the list when online so the row collapses away instead of leaving a zero-height wrapper; offline it stays put with the usual pending-delete styling.

One deliberate trade-off: a fully deleted group row now pops out rather than playing `FadeOutUp`, because it leaves the list on the same render the flag arrives. The surviving rows still slide up via `LinearTransition`, and the fade was mostly invisible anyway — `hideChildren` had already collapsed the row's content a step earlier.

### Fixed Issues

$ https://github.com/Expensify/App/issues/100122
PROPOSAL:

### Tests

- Delete all transactions in a group

1. Open Spend
2. Select any suggested group search
3. Expand to fetch all transactions
4. Click on the group checkbox/select each transaction
5. Delete selected transactions
6. Verify that the group disappears and doesn't flash again

- Delete all views transactions

1. Open Spend
2. Select any suggested group search
3. Set the limit result to less than the total expense in the group
4. Click on the group checkbox/select each transaction
5. Delete selected transactions
6. Verify that the group doesn't disappear and the group is collapsed

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as test

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/ef28a4aa-f9ae-4b1a-b43c-4b90dfa928c6

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/ba180f0e-4b1b-4d42-b67c-b370d56e60a4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/8be1a50e-b317-4960-b521-0449e0a786a9

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/6af4af55-0187-4c67-9e5b-9bff6b4ddd7f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/19b7bae1-99f6-44c9-9e9e-5a6551ed05b6

https://github.com/user-attachments/assets/f6127b6a-88d4-42f9-be42-423cca7c5820

https://github.com/user-attachments/assets/7799a3ec-f3d8-48ee-89f5-281c6aec0617

</details>

### CODE REVIEW

File: src/hooks/useSearchBulkActions.ts [hunk 1]
```diff
@@ -202,25 +204,57 @@ function isGroupSelection(key: string, transaction: SelectedTransactions[string]
...
+    return [...groupKeys];
+}
+
+function addSelectedGroupsFilter(queryJSON: SearchQueryJSON, selectedTransactions: SelectedTransactions, searchData: SearchResultDataType | undefined): SearchQueryJSON {
+    const {groupBy} = queryJSON;
+    if (!groupBy || !searchData) {
+        return queryJSON;
+    }
+
+    const groupKeys = getSelectedGroupKeys(selectedTransactions, searchData);
```
[the remark this answers is not in the document]
Comment (nkdengineer) [reply src/hooks/useSearchBulkActions.ts]: Makes sense, updated to use a separate util.

File: src/hooks/useDeleteTransactions.ts [hunk 2]
```diff
@@ -402,6 +405,8 @@ function useDeleteTransactions({report, reportActions, policy}: UseDeleteTransac
                     isSingleTransactionView,
                     transactionIDsPendingDeletion: deletedTransactionIDs,
                     selectedTransactionIDs: transactionIDs,
+                    searchHash: currentSearchHash,
+                    fullyDeletedGroupKeys,
```
[the remark this answers is not in the document]
Comment (nkdengineer) [reply src/hooks/useDeleteTransactions.ts]: If we don't go to `deleteMoneyRequest`. That means all transactions are split transactions; then the group is just collapsed and disappears after `Search` API. That means there's no flash anymore.

File: src/libs/actions/IOU/DeleteMoneyRequest.ts [hunk 3]
```diff
@@ -1091,6 +1110,15 @@ function deleteMoneyRequest({
         reportActionID: reportAction.reportActionID,
     };
 
+    // A group row in a grouped search outlives its child transactions, so flagging the group's own snapshot entry
+    // is what keeps the row out of the list until the next Search response drops it. Riding along with this request
+    // is what puts the row back if the delete fails.
+    const groupPendingDeleteData = getGroupPendingDeleteOnyxUpdate(searchHash, fullyDeletedGroupKeys ?? []);
```
[the remark this answers is not in the document]
Comment (nkdengineer) [reply src/libs/actions/IOU/DeleteMoneyRequest.ts]: Updated.

**Comment by cretadn22** [comment]
@nkdengineer The flicker still happens when deleting a group's last expense through its own thread

https://github.com/user-attachments/assets/23e4e81c-b4ec-4680-b6a4-aed6989a5375

**Comment by cretadn22** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified that the composer does not automatically focus or open the keyboard on mobile unless explicitly intended. This includes checking that returning the app from the background does not unexpectedly open the keyboard.
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: HybridApp
    - [x] Android: mWeb Chrome
    - [x] iOS: HybridApp
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: HybridApp</summary>



https://github.com/user-attachments/assets/f0e15995-d98c-49c0-a644-b3cc21525381



</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/a70ea1fa-c993-42b1-a9ff-c1f7043dc5d3




</details>

<details>
<summary>iOS: HybridApp</summary>



https://github.com/user-attachments/assets/67ab71cc-e125-41ed-b76a-c931092736f0



</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/619873be-9c1a-445e-a65e-97483e8e3c65


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/dd7e71b6-9c0b-46ed-959a-a4a626f7fb5a


</details>

**Comment by cretadn22** [comment]
I only noticed one bug (in [this comment](https://github.com/Expensify/App/pull/100675#issuecomment-5645228428)), the rest of the code changes are solid and testing well.

**Comment by nkdengineer** [comment]
@cretadn22 I fixed this issue.

File: src/components/Search/SearchList/ListItem/GroupHeader.tsx [hunk 4]
```diff
@@ -195,11 +196,7 @@ function GroupHeader({
         onCheckboxPress(withOriginalKey(item), isExpenseReportType ? undefined : groupItem.transactions);
     };
 
-    const pendingAction =
-        item.pendingAction ??
-        (groupItem.transactions.length > 0 && groupItem.transactions.every((transaction) => isTransactionPendingDelete(transaction))
-            ? CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE
-            : undefined);
+    const groupCount = getSearchGroupCount(groupItem);
```
Comment (cretadn22) [inline src/components/Search/SearchList/ListItem/GroupHeader.tsx]:
```suggestion
```

Comment (cretadn22) [reply src/components/Search/SearchList/ListItem/GroupHeader.tsx]: @nkdengineer The failing ESLint check is currently blocking us from moving forward

**Comment by cretadn22** [comment]
@nkdengineer Please check again

**Comment by nkdengineer** [comment]
@cretadn22 Updated.

**Review by cretadn22 — APPROVED** [review]

File: src/components/Search/selectionBuilders.ts [hunk 5]
```diff
@@ -1,16 +1,87 @@
...
+ * (expense-report rows), the loaded children are treated as the whole group.
+ */
+function isSelectionCoveringEntireGroup(groupCount: number | undefined, selectedCount: number, loadedSelectableCount: number): boolean {
+    if (selectedCount <= 0) {
+        return false;
+    }
+    if (groupCount === undefined) {
+        return loadedSelectableCount > 0 && selectedCount === loadedSelectableCount;
+    }
+    return selectedCount === groupCount;
```
Comment (neil-marcellini) [inline src/components/Search/selectionBuilders.ts:44]: I think there might be an issue here as noted by an AI.

---------

The group's `count` is never decremented optimistically, so I think a group deleted in two batches misses the flag on the second batch.

[`isSelectionCoveringEntireGroup`](https://github.com/Expensify/App/blob/fba8529df9e227ac4f99ce56e2177b7bde629c46/src/components/Search/selectionBuilders.ts#L37-L45) compares two numbers that come from different points in time:

- `selectedCount` counts the selection's children, and pending-delete children never get into the selection — the [group toggle filters them out](https://github.com/Expensify/App/blob/fba8529df9e227ac4f99ce56e2177b7bde629c46/src/components/Search/SearchWriteActionsProvider.tsx#L523) before building entries, and the same filter feeds `loadedSelectableCount`.
- `groupCount` is the snapshot's `count`, which still includes them. The create path [increments `count` optimistically](https://github.com/Expensify/App/blob/fba8529df9e227ac4f99ce56e2177b7bde629c46/src/libs/actions/IOU/SearchUpdate.ts#L212-L221) when an expense is added to a `groupBy:from` search, but nothing decrements it on delete — `getGroupPendingDeleteOnyxUpdate` only writes `pendingAction`.

So with a 3-expense group, after one expense is pending delete: selecting the other two gives `selectedCount` 2 against `groupCount` 3, the equality fails, and the group is left out of `getSelectedGroupKeys`. [The same comparison in `getSelectedGroupKeys`](https://github.com/Expensify/App/blob/fba8529df9e227ac4f99ce56e2177b7bde629c46/src/hooks/useSearchBulkActions.ts#L234-L245) has it too.

Offline this is deterministic, and it's a behavior change rather than just a missed optimization: every expense in the group is now pending delete, but the group header renders with no pending-delete styling, because [the children-derived fallback that used to cover exactly this case](https://github.com/Expensify/App/blob/07dca732efa6a18f69f1bdeda7bff08b87161e44/src/components/Search/SearchList/ListItem/GroupHeader.tsx#L198-L202) is removed in this PR. Online it's a race against the `Search` refresh that resets `count` — probably invisible most of the time, which is what makes me want it handled rather than left to timing.

Two ways I can see to fix it, and I'd lean toward the second:

1. Subtract the group's pending-delete children from `groupCount` before comparing, so both sides describe the same moment. The call sites already compute the pending-delete split, so `loadedChildren.length - loadedSelectableCount` is available at each one.
2. Decrement `count` in the delete's `optimisticData` (and restore it in `failureData`), mirroring what the create path does.

### Test steps to confirm

1. Open a grouped expense search (e.g. `group-by:merchant`) with a group holding 3 expenses.
2. Go offline.
3. Select one expense in that group and delete it.
4. Select the remaining two and delete them.
5. Expected: the group header greys out like its children. Actual (I believe): the children grey out, the header doesn't.

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
Wow this is a big one. Nice job. I'm requesting changes for one likely bug that AI found. Everything else looks good. Two more minor non-blocking comments from AI that you could clean up.

--------

### C3 — PR description describes a fallback that isn't in the code

**Non-blocking.** The description says "The wide-web header still has a children-derived pending-delete fallback, but only when the loaded children cover the whole group." `GroupHeader` now reads [`item.pendingAction`](App-pr-100675/src/components/Search/SearchList/ListItem/GroupHeader.tsx#L349) only — the fallback is fully removed. Ask for the description to be corrected, since reviewers and QA work from it.

### C4 — Comment contradicts the code it sits above

**Non-blocking.** [ExpenseGroupedSearchView.tsx:124](App-pr-100675/src/components/Search/ExpenseGroupedSearchView.tsx#L124) says the row is dropped "and let its exit animation play." The [filter on the next line](App-pr-100675/src/components/Search/ExpenseGroupedSearchView.tsx#L125) unmounts the row before `AnimatedExitRow` can render it, so nothing plays — which the PR body itself acknowledges as a deliberate trade-off ("pops out rather than playing `FadeOutUp`"). The comment should say that.

File: src/components/Search/selectionBuilders.ts (hunk 5 above)
Comment (nkdengineer) [reply src/components/Search/selectionBuilders.ts:57]: Good catch, I fixed it with the first solution.

**Comment by nkdengineer** [comment]
@neil-marcellini All done.

File: tests/unit/Search/getSelectedGroupKeysTest.ts [hunk 6]
```diff
@@ -90,6 +90,20 @@ describe('getSelectedGroupKeys', () => {
 
         expect(keys).toEqual([]);
     });
+
+    it('returns a group whose remaining children are selected after some are pending delete', () => {
```
Comment (neil-marcellini) [inline tests/unit/Search/getSelectedGroupKeysTest.ts]: NAB: this test is kind of confusing because it describes a case that is not really set up in the test. We're kind of testing an intermediate behavior and really just testing that if a selection has its entire group selected, then we honor that (as opposed to checking the count within this function). The name doesn't really match what we're testing here.

Comment (neil-marcellini) [reply tests/unit/Search/getSelectedGroupKeysTest.ts]: I think the tests for selectionBuilder are much more clear so maybe we delete these and keep those.

**Review by neil-marcellini — DISMISSED** [review]
Good to go, thanks for the updates!


AI found another possible issue, but we can fix it in a follow up.

--------
One new issue, non-blocking. Making isEntireGroupSelected sufficient in getSelectedGroupKeys means a stale flag can now over-report. Reconcile [copies the flag forward verbatim](https://github.com/Expensify/App/blob/251f4e251307a4837a9bd4e57e26bfa143f32226/src/components/Search/SearchWriteActionsProvider.tsx#L221) without re-evaluating it against the new data. So: on group-by:from, select a whole 2-expense group, then create a third expense that lands in that group (the create path bumps count and adds the row, but doesn't touch the selection). The two originals keep isEntireGroupSelected: true, delete flags the whole group, and the surviving third expense's group row vanishes until the next Search response. Before this commit the live selectedCount === groupCount check was self-correcting here.

Narrow, and the same staleness already existed for the unknown-count branch, so I'd treat it as a follow-up rather than a blocker. The clean fix is for reconcile to recompute the flag — it already has the group's transactions and count in hand at that point.

File: tests/unit/Search/getSelectedGroupKeysTest.ts (hunk 6 above)
Comment (nkdengineer) [reply tests/unit/Search/getSelectedGroupKeysTest.ts]: @neil-marcellini I removed these tests and fixed the edge case.

**Review by neil-marcellini — APPROVED** [review]
Thanks for the update!

Two more NAB comments from the AI. They can be addressed in a follow up. The first one is about performance so maybe not a problem unless we prove it's actually non-performant and under which conditions.

--------

stampGroupCoverageFlags is called once per group with the entire accumulated newTransactionList, and it spreads and scans that object twice each time. With a select-all across many groups this goes quadratic in a hook that reruns on every filteredData change. The block already knows the group's own children, so computing the flag once and assigning it to those entries would avoid the whole-selection scans.

Line 219 deliberately leaves groupKey undefined for isExpenseReportType, and [the new loop then sets it anyway](https://github.com/Expensify/App/blob/92da38ae6ed8eff26cdb4c9098caaaafe06d75a5/src/components/Search/SearchWriteActionsProvider.tsx#L238) for every group type. Downstream consumers all gate on the group_ prefix so I can't find a behavior change, but it reads like the guard was undone by accident — worth a word from the author on whether that's intended.

**Comment by neil-marcellini** [comment]
Dang, since I authored some of the commits on this PR, we need another reviewer.

**Comment by neil-marcellini** [comment]
(Neil's AI agent)

@Beamanator you've been assigned as an independent reviewer here via PullerBear.

PullerBear wasn't working on this PR, so I ran it on a dummy PR in Web-Expensify instead and carried the result over: https://github.com/Expensify/Web-Expensify/pull/50616 (PullerBear picked Beamanator there, and that PR is now closed again).

Context on why another reviewer is needed: the `Verify peer review` check fails because I'm a commit author on this PR as well as an approver, so my approval doesn't count as independent.

**Review by Beamanator — APPROVED** [review]
LGTM
